lib/mixlib/shellout.rb

```diff
@@ -297,7 +289,6 @@ def parse_options(opts)
           self.password = setting
         when 'user'
           self.user = setting
-          self.with_logon = setting
```

Comment:
This isn't only used by core chef, its found its way into lots of stuff.  I'd treat it as a public API, particularly since someone took the trouble to declare with_login as an accessor, so you have to assume its getting used out in the wild.

The uid and gid issue should also get preserved, the argument that someone might be using them is a lot weaker, but I'd just mark them as to be deprecated on windows for now with a comment rather than worrying about it.